src/main/java/bowling/domain/FinalFrame.java

```diff
@@ -0,0 +1,79 @@
+package bowling.domain;
+
+import java.util.Objects;
+
+public class FinalFrame implements Frame {
+    private static final int FRAME_MAX_LENGTH = 3;
+    public static final int MAX_PIN_SUM = 10;
+
+    private Pin pin;
+    private final States states;
+
+    public FinalFrame() {
+        this.pin = new Pin(Pin.MIN_PIN);
+        this.states = new States();
+    }
+
+    @Override
+    public void bowl(Pin fallenPin) {
+        State state = State.finalBowl(this.pin.getFallenPin(), fallenPin.getFallenPin(), this.states.getLastState());
+        setStates(state);
+        setPin(fallenPin);
+    }
+
+    @Override
+    public boolean isEndFrame() {
+        return this.states.getSize() > FRAME_MAX_LENGTH;
+    }
+
+    @Override
+    public boolean isEndGame() {
+        if (states.getSize() == NormalFrame.FRAME_MAX_LENGTH && getStatesPinSum() < MAX_PIN_SUM) {
+            return true;
+        }
+        return this.states.getSize() == FRAME_MAX_LENGTH;
+    }
+
+    @Override
+    public Frame getNextFrame(int frameNumber) {
+        return null;
+    }
```

Comment:
null을 반환하는것보다 요청한 오퍼레이션을 지원하지 않는다는 예외를 내뱉는것에 대해선 어떻게 생각하시나요 :) ?
```suggestion
    public Frame getNextFrame(int frameNumber) {
        throw new UnsupportedOperationException();
    }
```

Reply:
이런 방법이 있었네요! 감사합니다 👍